packages/react/src/primitives/utils/useStableId.ts

```diff
@@ -0,0 +1,31 @@
+// Adapted from https://github.com/radix-ui/primitives/blob/main/packages/react/id/src/id.tsx#L8
+import * as React from 'react';
+import { useLayoutEffect } from './useLayoutEffect';
+
+// Prefixed autogenerated id created by useStableId
+export const AUTO_GENERATED_ID_PREFIX = 'amplify-id';
+
+export type UseStableId = (id?: string) => string;
+
+// Create a local version of React.useId which will reference React.useId for React 18
+// and fallback to noop for React 17 and below
+const useReactId = (React as any).useId || (() => undefined);
+let count = 0;
+
+/**
+ * Create a uuid to use with amplify fields unless
+ * an id is provided
+ * @param id user specified id
+ * @returns string
+ */
+export const useStableId: UseStableId = (id) => {
```

Comment:
if `id` is provided, can we return it asap?

Reply:
We can't early return before the hooks or we won't be following the rules of hooks. I don't think we can return it any earlier.

Reply:
right, resolving this comment